.changeset/tidy-flies-cheer.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/config': patch
+---
+
+Handle a case when boolean configuration parameters are given a string type of 'true' or 'false'. This happens particularly when such parameters are used with environmental substitution as environment variables are always strings.
```

Comment:
Think this changeset could do with adding a few more of the supported boolean types now as we've added some more, maybe it makes sense to add these to the list too.

```suggestion
Adds the ability to be able to coerce values to their boolean representatives. 
Thinks such as `"true"` `1` `on` and `y` will become `true` when using `getBoolean` and the opposites `false`.
This happens particularly when such parameters are used with environmental substitution as environment variables are always strings.
```